run.go

```diff
@@ -14,11 +14,29 @@ import (
 	"github.com/cyverse-de/road-runner/fs"
 	"github.com/kr/pty"
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
 	"github.com/spf13/viper"
 	"gopkg.in/cyverse-de/messaging.v4"
 	"gopkg.in/cyverse-de/model.v2"
 )
 
+// logrusProxyWriter will prevent
+// "Error while reading from Writer: bufio.Scanner: token too long" errors
+// if a docker command generates a lot of output
+// (from pulling many input containers at once, for example)
+// and Logrus attempts to log all of that output in one log line.
+type logrusProxyWriter struct {
+	entry *logrus.Entry
+}
+
+func (w *logrusProxyWriter) Write(b []byte) (int, error) {
+	return fmt.Fprintf(w.entry.Writer(), string(b))
```

Comment:
I'm not exactly sure why `logrus` was logging all the `docker pull` output into 1 log entry before, but proxying through `fmt.Fprintf` (and also converting the bytes to a `string` first) seems to break up the logged output into 1 line for each newline in the `docker pull` output.